src/store/mutations.ts

```diff
@@ -179,6 +189,30 @@ const mutations: MutationTree<CalChartState> = {
   setGrapherToolDots(state, grapherToolDots: StuntSheetDot[]): void {
     state.grapherToolDots = grapherToolDots;
   },
+
+  // selection
+  clearSelectedDots(state): void {
+    state.selectedDots = [];
+  },
+  addSelectedDots(state, dots: number[]): void {
+    state.selectedDots = state.selectedDots
+      .concat(dots)
+      .filter((v, i, a) => a.indexOf(v) === i);
```

Comment:
Instead of adding the dots, then filtering, how about doing a for loop over `dots` and add if it is not in `selectedDots`?

```suggestion
    dots.forEach((dot) => state.selectedDots.indexOf(dot) !== -1 && state.selectedDots.push(dot))
```

Reply:
ok